### ISSUE TITLE
`shadow` classes incompatible with ring utilities

### ISSUE THREAD

**Issue by Dramloc** [ben-rogerson/twin.macro#224]
Hi,

I found an issue when using `shadow-*` classes with the new ring utilities. Using one seems to override the other, making them incompatible.

Here are some examples:
```js
{/* The ring classes will override `shadow-lg` (we only see the ring) */}
<div tw="shadow-lg ring-1 ring-black ring-opacity-5" />
```
```js
{/* `shadow-lg` will override the ring classes (we only see the shadow) */}
<div tw="ring-1 ring-black ring-opacity-5 shadow-lg" />
```

You can find reproduction examples here: https://codesandbox.io/s/twinmacro-ring-and-shadow-issue-1p99p.

Implementing the same behavior as tailwind could fix this issue. In v2, `shadow-*` utilities now create a `--tw-shadow` CSS variable and a `box-shadow` using shadow and ring variables instead of a single `box-shadow` (see https://tailwindcss.com/docs/box-shadow).

Thanks again for your time 🤗

**Comment by ben-rogerson** [comment]
Good catch and thank you for that demo.

I'll patch that up shortly 👍

**Comment by ben-rogerson** [comment]
Ok all fixed in v2.0.4, cheers
